### PR TITLE
bpo-39674: Revert "bpo-25988: Do not expose abstract collection classes in the collections module. (GH-10596)"

### PR DESCRIPTION
**Pull request by vstinner**
This reverts commit ef092fe9905f61ca27889092ca1248a11aa74498.

Update collections __getattr__() and documentation to defer aliases
removal to Python 3.10.

https://bugs.python.org/issue39674

### CODE REVIEW

**Comment by vstinner** [comment]
Since https://bugs.python.org/issue39674 is controversial, I would prefer to get at least one formal approval of this revert from core developers.

cc @brettcannon @warsaw @serhiy-storchaka @gvanrossum @zooba

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. I do not care about other removals, but this change looks affecting too much code.

**Comment by vstinner** [comment]
Data point from Fedora packages: around 65 packages failed to build because of commit ef092fe9905f61ca27889092ca1248a11aa74498 which removed the collections aliases.

See Miro's email: https://mail.python.org/archives/list/python-dev@python.org/message/LGGI42UEQMJVNW2ZHEP473Z7ZQ5BDBPL/

This revert should fix most of these packages. "most" because we don't know if packages build will fail for another reasons (maybe another deprecated function removed from Python 3.9).

**Comment by vstinner** [comment]
Fedora packages reported failures to most packages upstream and for some of them, we even proposed fixes.

**Comment by vstinner** [comment]
The macOS failure is: https://bugs.python.org/issue38691

**Review by gvanrossum — APPROVED** [review]

**Comment by vstinner** [comment]
Merged, thanks for the approval @serhiy-storchaka and @gvanrossum.

**Comment by terryjreedy** [comment]
Shouldn't '3.8' in
"For backwards compatibility, they continue to be visible in this module through Python 3.8."
be changed to '3.9'?

FWIW, I also think this 1 reversion was the best thing to do and perhaps had more support than revert nothing and revert multiple things.

**Comment by vstinner** [comment]
> Shouldn't '3.8' in "For backwards compatibility, they continue to be visible in this module through Python 3.8." be changed to '3.9'?

Right: I created https://github.com/python/cpython/pull/18747 to update the doc.

**Pull request merged**
